sklearn/utils/validation.py

```diff
@@ -46,26 +46,33 @@ class NotFittedError(_NotFittedError):
 warnings.simplefilter('ignore', _NonBLASDotWarning)
 
 
-def _assert_all_finite(X):
+def _assert_all_finite(X, allow_nan):
```

Comment:
(i.e if you are successively removing values from your dataset for benchmarking... And need to check the previous step for any `inf`)

Basically this was a part of #5974. I moved it here as this would be merged first.